src/ReleaseNotesUtil/Program.cs

```diff
@@ -0,0 +1,252 @@
+                    }
+                }
+            }
+
+            Dictionary<string, RuleInfo> oldRulesById = oldContent.Rules.ToDictionary(r => r.Id);
+            Dictionary<string, RuleInfo> newRulesById = newContent.Rules.ToDictionary(r => r.Id);
+            IEnumerable<RuleInfo> addedRules =
+                newContent.Rules
+                    .Where(r => !oldRulesById.ContainsKey(r.Id));
+            IEnumerable<RuleInfo> removedRules =
+                oldContent.Rules
+                    .Where(r => !newRulesById.ContainsKey(r.Id));
+            StringBuilder sb = new StringBuilder();
+            GenerateRulesDiffMarkdown(sb, "### Added", addedRules);
+            GenerateRulesDiffMarkdown(sb, "### Removed", removedRules);
+            File.WriteAllText(outputPath, sb.ToString());
+        }
+
+        private static void WriteRuleFileContent(RuleFileContent ruleFileContent, string outputPath)
+        {
+            DataContractJsonSerializer serializer = new DataContractJsonSerializer(typeof(RuleFileContent));
+            using (FileStream fs = new FileStream(outputPath, FileMode.Create, FileAccess.Write, FileShare.None))
+            {
+                serializer.WriteObject(fs, ruleFileContent);
+            }
+        }
+
+        private static RuleFileContent ReadRuleFileContent(string path)
+        {
+            DataContractJsonSerializer serializer = new DataContractJsonSerializer(typeof(RuleFileContent));
+            using (FileStream fs = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read))
+            {
+                return (RuleFileContent)serializer.ReadObject(fs);
+            }
+
+        }
+
+        private static void GenerateRulesDiffMarkdown(StringBuilder sb, string heading, IEnumerable<RuleInfo> rules)
+        {
+            if (!rules.Any())
+            {
+                return;
+            }
+
+            IEnumerable<RuleInfo> sortedRules = rules.OrderBy(r => r, CategoryThenIdComparer.Instance);
+
+            sb.AppendLine(heading);
+            string previousCategory = null;
+            foreach (RuleInfo rule in sortedRules)
+            {
+                if (rule.Category != previousCategory)
+                {
+                    sb.AppendLine($"- {rule.Category}");
+                    previousCategory = rule.Category;
+                }
+
+                sb.AppendLine($"  - {rule.IdWithHelpLinkMarkdown}: {rule.Title}{(rule.IsEnabledByDefault ? " -- **Enabled by default**" : "")}");
+            }
+        }
+
+        private static IEnumerable<string> GetFxCopAnalyzerBinaries(string nugetInstalledPackagesPath, string version)
+        {
+            if (!Directory.Exists(nugetInstalledPackagesPath))
+            {
+                throw new ArgumentException($"'{nugetInstalledPackagesPath}' is not a directory or doesn't exist");
+            }
+
+            string[] roslynAnalyzerPackages = new string[] {
+                "Microsoft.CodeQuality.Analyzers",
```

Comment:
Hard coding these values are concerning, as in the past we have changed the package names and/or moved them around quite a bit. Hopefully, it is stabilized now, but say like we decide to add a new sub-package, then it won't be intuitive to figure out that this list needs to be updated.

Reply:
Yep, definitely less than ideal.  When I'm feeling more ambitious, I was thinking of going through the nuget package's dependencies.